### DISCUSSION TITLE
Missing some vector functions?

### DISCUSSION THREAD

**Discussion by Tennyleaz** [fabricjs/fabric.js#9645] [Q&A]
Hi,
I am using `fabric` 5.3.0 and `types/fabric` 5.3.6 with typescript.
I found some inconsistent of vector functions in `fabric.utils`, between this repo and official document, and the downloaded packages.

For example I cannot find [fabric.util.getBisector](http://fabricjs.com/docs/fabric.util.html#.getBisector) and [fabric.util.getHatVector](http://fabricjs.com/docs/fabric.util.html#.getHatVector) in this repo's [vectors.ts](https://github.com/fabricjs/fabric.js/blob/master/src/util/misc/vectors.ts), and not in `types/fabric`. However they will work in runtime!
Another example, I see `Point.rotate` function in this repo's [Point.ts](https://github.com/fabricjs/fabric.js/blob/master/src/Point.ts#L13), but not in the [official document](http://fabricjs.com/docs/fabric.Point.html), nor in the downloaded npm package.

How do I know which source is the right place to check?

**Comment by Smrtnyk** [comment]
is types/fabric maintained by fabricjs core team?
if not, then I wouldn't rely on it being 100% accurate

**Comment by ShaMan123** [comment]
It is not maintained by us
As written in #8299 you should remove it and us our built in ones.

**Reply by ShaMan123** [reply]
I told them to remove it
